src/frontend/Sider/Behandling/Vilkårvurdering/oppdatering.ts

```diff
@@ -0,0 +1,54 @@
+import { delvilkårSomErRelevante } from './utils';
+import { RegelId, VurderingInput, SvarId, Delvilkårsett, Delvilkår } from '../vilkår';
+
+export type OppdaterVilkårsvurdering = {
+    id: string;
+    behandlingId: string;
+    vurdering: OppdaterDelvilkårvurdering[];
+};
+
+interface OppdaterDelvilkårvurdering {
+    regel: RegelId;
+    svar: SvarId | null;
+    begrunnelse: string | null;
+}
+
+const skalHaBegrunnelse = (delvilkårsvurdering: Delvilkår): boolean =>
+    delvilkårsvurdering.svar != null &&
+    delvilkårsvurdering.svaralternativer[delvilkårsvurdering.svar].begrunnelsestype !== 'UTEN';
+
+export const mapTilOppdaterDelvilkårsvurderinger = (
+    vilkårsvurdering: Delvilkårsett
+): OppdaterDelvilkårvurdering[] =>
+    delvilkårSomErRelevante(vilkårsvurdering).map(([regelId, delvilkårsvurdering]) => ({
+        regel: regelId,
+        svar: delvilkårsvurdering.svar,
+        begrunnelse: skalHaBegrunnelse(delvilkårsvurdering)
+            ? delvilkårsvurdering.begrunnelse
+            : null,
```

Comment:
Hvorfor trengs denne?

Reply:
Fordi begrunnelse i React-staten lever selv om regelen skjules. Det er ønskelig, fordi da kan brukeren toggle mellom svar uten å risikere å miste begrunnelser. Men ref diskusjonen om ikke å sende data brukeren ikke ser, så må vi fjerne begrunnelsen før vi sender til backend. 

Jeg gjorde en renaming for forhåpentligvis å gjøre det mer tydelig.